apps/web/src/app/vault/vault-filter/services/vault-filter.service.ts

```diff
@@ -0,0 +1,285 @@
+
+  constructor(
+    protected stateService: StateService,
+    protected organizationService: OrganizationService,
+    protected folderService: FolderService,
+    protected cipherService: CipherService,
+    protected collectionService: CollectionService,
+    protected policyService: PolicyService,
+    protected i18nService: I18nService
+  ) {
+    this.loadSubscriptions();
+  }
+
+  protected loadSubscriptions() {
+    this.folderService.folderViews$
+      .pipe(
+        combineLatestWith(this._organizationFilter),
+        switchMap(([folders, org]) => {
+          return this.filterFolders(folders, org);
+        }),
+        takeUntil(this.destroy$)
+      )
+      .subscribe(this._filteredFolders);
+
+    // TODO: Use collectionService once collections is refactored
+    this.collectionViews$
+      .pipe(
+        combineLatestWith(this._organizationFilter),
+        switchMap(([collections, org]) => {
+          return this.filterCollections(collections, org);
+        }),
+        takeUntil(this.destroy$)
+      )
+      .subscribe(this._filteredCollections);
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+
+  // TODO: Remove once collections is refactored with observables
+  async reloadCollections() {
+    this.collectionViews$.next(await this.collectionService.getAllDecrypted());
+  }
+
+  // TODO: Remove once organizations is refactored with observables
+  async reloadOrganizations() {
+    this._organizations.next(await this.organizationService.getAll());
+  }
+
+  async storeCollapsedFilterNodes(collapsedFilterNodes: Set<string>): Promise<void> {
+    await this.stateService.setCollapsedGroupings(Array.from(collapsedFilterNodes));
+    this._collapsedFilterNodes.next(collapsedFilterNodes);
+  }
+
+  protected async buildCollapsedFilterNodes(): Promise<Set<string>> {
+    const nodes = new Set(await this.stateService.getCollapsedGroupings());
+    this._collapsedFilterNodes.next(nodes);
+    return nodes;
+  }
+
+  updateOrganizationFilter(organization: Organization) {
+    if (organization?.id != "AllVaults") {
+      this._organizationFilter.next(organization);
+    } else {
+      this._organizationFilter.next(null);
+    }
+  }
+
+  async expandOrgFilter() {
+    const collapsedFilterNodes = await this.buildCollapsedFilterNodes();
+    if (!collapsedFilterNodes.has("AllVaults")) {
+      return;
+    }
+    collapsedFilterNodes.delete("AllVaults");
+    await this.storeCollapsedFilterNodes(collapsedFilterNodes);
+  }
+
+  protected async buildOrganizationTree(
+    orgs?: Organization[]
+  ): Promise<TreeNode<OrganizationFilter>> {
+    const headNode = this.getOrganizationFilterHead();
+    if (!(await this.checkForPersonalOwnershipPolicy())) {
+      const myVaultNode = this.getOrganizationFilterMyVault();
+      headNode.children.push(myVaultNode);
+    }
+    if (await this.checkForSingleOrganizationPolicy()) {
+      orgs = orgs.slice(1);
```

Comment:
Oops, off-by-one! My mistake 😁 

```suggestion
      orgs = orgs.slice(0, 1);
```

to get the first item.